### PR TITLE
type annotation for str.translate

### PR DESCRIPTION
**Pull request by GrozniyToaster**
by issue [#8628](https://github.com/python/typeshed/issues/8628)

### CODE REVIEW

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -535,7 +538,7 @@ class str(Sequence[str]):
     def title(self: LiteralString) -> LiteralString: ...
     @overload
     def title(self) -> str: ...  # type: ignore[misc]
-    def translate(self, __table: Mapping[int, int | str | None] | Sequence[int | str | None]) -> str: ...
+    def translate(self, __table: _TranslateTable) -> str: ...
```
Comment (sobolevn) [inline stdlib/builtins.pyi:541]: Or maybe `_typeshed.SupportsGetItem`? https://github.com/python/typeshed/blob/0259068ad6bc814a510009cade00fd961b5d0355/stdlib/_typeshed/__init__.pyi#L121

Note, that variance there is different.

Comment (sobolevn) [reply stdlib/builtins.pyi:541]: I had this example in mind:

```python
>>> class Some:
...    def __getitem__(self, key):
...        return MyInt(1)
... 
>>> 'a'.translate(Some())
'\x01'
```

Comment (sobolevn) [reply stdlib/builtins.pyi:541]: Oh, `SupportsGetItem` is also a `Container`, never mind.

**Review by Akuli — APPROVED** [review]
Thanks! The example from your issue now passes the type check, even when modified to return a custom subclass of `int`:

```python
from __future__ import annotations
from typing import Final
class MyInt(int): pass

class SeparatorTranslator:
    separators_whitelist: Final = {' ', '\n', '\t'}

    def __getitem__(self, char_order_index: int) -> str | MyInt:
        char = chr(char_order_index)
        if char.isspace() and char not in self.separators_whitelist:
            return MyInt(ord(' '))
        else:
            return char

print('asd asd \v\t asd'.translate(SeparatorTranslator()))
```

**Comment by Akuli** [comment]
Presumably `bytes.translate()` and `bytearray.translate()` could be updated too, but they seem to be different from `str.translate()` and can be done in a separate PR.

**Pull request merged**